### ISSUE TITLE
[DX] Fix the documentation about the Symfony security

### ISSUE THREAD

**Issue by javiereguiluz** [symfony/symfony-docs#4158]
If you read [this DX issue](https://github.com/symfony/symfony/issues/11728) regarding the security roles, you can read one of the usual awesome replies by @stof: https://github.com/symfony/symfony/issues/11728#issuecomment-52912608

You'll see that the Security chapter and cookbooks need some serious improvements. Given that we are talking about Security, I think that this should be a high priority issue.

For now, this is the roadmap of changes to make:
- [ ] Never mention that `IS_AUTHENTICATED_*` are roles. They are not.
- [ ] Better explain the differences between roles, strings, expressions, ACL permissions maps, etc.
- [ ] ...

**Comment by iltar** [comment]
I ended up in here when trying to find out what the token methods did [`setAttributes`, `getAttributes` and ` hasAttribute`](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Security/Core/Authentication/Token/AbstractToken.php#L166-L196). As I can see, there are no default implementations within symfony and there's no documentation mentioning those specific methods. 

@stof So what is their intention? What are they supposed to be used for? If they are not for Authorization, what's their Authentication intent? You mentioned they are used inside voters, but that's Authorization.

**Comment by javiereguiluz** [comment]
We agree that the Symfony Security docs need a lot of improvements. This is in our priority list ... but it's taking us a lot of time because of the massive complexity of the Security internals.

In any case, to focus all our discussions about this in one place, we've created a meta issue in #7496 and we've linked this issue from there. That's why we're closing your issue ... but only to avoid duplicated discussions. We won't forget about what you said here. Thanks!
